presto-password-authenticators/src/main/java/io/prestosql/plugin/password/salesforce/SalesforceConfig.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.password.salesforce;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.log.Logger;
+
+import javax.validation.constraints.Max;
+import javax.validation.constraints.NotNull;
+
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Set;
+
+public class SalesforceConfig
+{
+    private static final Logger log = Logger.get(SalesforceConfig.class);
+
+    protected static final String LOGINURL = "https://login.salesforce.com/services/Soap/u/";
+    protected static final String APIVERSION = "46.0";
+
+    protected static final String LOGIN_SOAP_MESSAGE = "<?xml version=\"1.0\" encoding=\"utf-8\" ?>\n" +
+            "<env:Envelope xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\"\n" +
+            "xmlns:urn=\"urn:enterprise.soap.sforce.com\"\n" +
+            "   xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\n" +
+            "   xmlns:env=\"http://schemas.xmlsoap.org/soap/envelope/\">\n" +
+            " <env:Header>\n" +
+            "     <urn:CallOptions>\n" +
+            "       <urn:client>presto</urn:client>\n" +
+            "     </urn:CallOptions>\n" +
+            " </env:Header>\n" +
+            " <env:Body>\n" +
+            "   <n1:login xmlns:n1=\"urn:partner.soap.sforce.com\">\n" +
+            "     <n1:username>%s</n1:username>\n" +
+            "     <n1:password>%s</n1:password>\n" +
+            "   </n1:login>\n" +
+            " </env:Body>\n" +
+            "</env:Envelope>\n";
+
+    public static final int MAX_EXPIRE = 3600;
```

Comment:
This can be private

Reply:
Originally didn't depend on the annotations for validation, and therefore used this throughout the package.  I've refactored to use `@MaxDuration` per your suggestion.